File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {

        Set<Integer> newSet = new HashSet<>();
        int result = 0;

        for(int i = 0; i< nums.length; i++){
            if(newSet.contains(nums[i])){
                result = nums[i];
            }
            else{
                newSet.add(nums[i]);
            }
        }

        return result;

        
    }
}

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:

        new = set()
        i = 0

        while i < len(nums):
            if nums[i] not in new:
                new.add(nums[i])
            else:
                return nums[i]

            i = i+1


          

        